lldb/source/Plugins/SymbolFile/DWARF/DWARFUnit.cpp

```diff
@@ -360,8 +364,10 @@ void DWARFUnit::SetDwoStrOffsetsBase() {
     const DWARFDataExtractor &strOffsets =
         GetSymbolFileDWARF().GetDWARFContext().getOrLoadStrOffsetsData();
     uint64_t length = strOffsets.GetU32(&baseOffset);
-    if (length == 0xffffffff)
+    if (length == llvm::dwarf::DW_LENGTH_DWARF64) {
       length = strOffsets.GetU64(&baseOffset);
+      m_str_offset_size = 8;
+    }
```

Comment:
Do we have a utility for reading the 32/64 encoded length we could reuse here?

Reply:
I was able to use the LLVM DWARFDataExtractor and re-use the `getInitialLength()` function from there.